test/built-ins/Iterator/concat/throws-typeerror-when-iterable-not-an-object.js

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2024 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-iterator.concat
+description: >
+  Throws a TypeError when the iterable is not an object.
+info: |
+  Iterator.concat ( ...items )
+
+  1. Let iterables be a new empty List.
+  2. For each element item of items, do
+    a. If item is not an Object, throw a TypeError exception.
```

Comment:
Relatedly, the primitive object wrappers (`new Boolean(true)`, etc) should pass this step. We should have a test for that.